sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java

```diff
@@ -563,6 +586,20 @@ public void deleteContact(String contactUuid) {
 					});
 				}
 
+				// Load latest events info
+				// Adding a second query here is not perfect, but selecting the last event with a criteria query
+				// doesn't seem to be possible and using a native query is not an option because of user filters
+				List<ContactEventSummaryDetails> eventSummaries = eventService.getEventSummaryDetailsByContacts(resultContactsUuids);
```

Comment:
Move this line to above the for loop; right now this is executed once for every contact in the list